plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManagerUtils.java

```diff
@@ -0,0 +1,120 @@
+package io.jenkins.tools.pluginmanager.impl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class PluginManagerUtils {
+
+    private PluginManagerUtils() {
+    }
+
+    /**
+     * Convert object arguments to strings and pass through to the String-ish method to append the path section
+     * onto the base URL string
+     *
+     * @see #appendPathOntoUrl(String, String...)
+     */
+    static String appendPathOntoUrl(Object urlString, Object... pathSection) {
```

Comment:
object is a very general API, aren't you just passing URLs? here might be simpler to change the type?

Reply:
It works for a couple of object types. We just do a toString to cut down the clutter in the place where it is called.